core/src/main/java/org/apache/gravitino/connector/credential/PathBasedContext.java

```diff
@@ -0,0 +1,63 @@
+/*
+ *  Licensed to the Apache Software Foundation (ASF) under one
+ *  or more contributor license agreements.  See the NOTICE file
+ *  distributed with this work for additional information
+ *  regarding copyright ownership.  The ASF licenses this file
+ *  to you under the Apache License, Version 2.0 (the
+ *  "License"); you may not use this file except in compliance
+ *  with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing,
+ *  software distributed under the License is distributed on an
+ *  "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ *  KIND, either express or implied.  See the License for the
+ *  specific language governing permissions and limitations
+ *  under the License.
+ */
+
+package org.apache.gravitino.connector.credential;
+
+import org.apache.gravitino.annotation.DeveloperApi;
+
+/**
+ * The {@code PathBasedContext} class represents the path and its associated credential type to
+ * generate a credential for {@link org.apache.gravitino.credential.CredentialOperationDispatcher}.
+ */
+@DeveloperApi
+public class PathBasedContext {
+
+  private final String path;
+
+  private final String credentialType;
+
+  /**
+   * Constructs a new {@code PathBasedContext} instance with the given path and credential type.
+   *
+   * @param path The path string.
+   * @param credentialType The type of the credential.
+   */
+  public PathBasedContext(String path, String credentialType) {
```

Comment:
`PathBasedContext` is quite similar to `PathBasedCredentialContext`, can we have a better name, like `PathContext`?

Reply:
updated